From 8608b8c116ce813ce68c99986e40caaffc4d817b Mon Sep 17 00:00:00 2001
From: Santhosh Kumar <santhosh.kumar.selvaraj@qt.io>
Date: Mon, 15 Sep 2025 13:19:27 +0200
Subject: [PATCH 042/239] Docs: Correct the type of the ValueFilter::value
 property

The type of ValueFilter::value property been used with a variant type,
but the documentation incorrectly mentions it as string type. This patch
updates the correct type of this property.

Fixes: QTBUG-140143
Change-Id: I15141d256657f33e9e6c06cc3819f300bb9ab11c
Reviewed-by: Mitch Curtis <mitch.curtis@qt.io>
(cherry picked from commit 88ee904cc7877d6c3695df51b0ddbee374256f8c)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/qmlmodels/sfpm/filters/qqmlvaluefilter.cpp | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/src/qmlmodels/sfpm/filters/qqmlvaluefilter.cpp b/src/qmlmodels/sfpm/filters/qqmlvaluefilter.cpp
index 3d373fed9a..7d8aba3d15 100644
--- a/src/qmlmodels/sfpm/filters/qqmlvaluefilter.cpp
+++ b/src/qmlmodels/sfpm/filters/qqmlvaluefilter.cpp
@@ -45,11 +45,9 @@ QQmlValueFilter::QQmlValueFilter(QObject *parent) :
 }
 
 /*!
-    \qmlproperty string ValueFilter::value
+    \qmlproperty variant ValueFilter::value
 
-    This property holds specific value that can be used to filter the data.
-
-    The default value is empty string.
+    This property holds the specific value that can be used to filter the data.
 */
 const QVariant& QQmlValueFilter::value() const
 {
-- 
2.51.2

